WordPress/gutenberg
# Add stylelint rule to disallow the `order` CSS property

Fixes https://github.com/WordPress/gutenberg/issues/61241

## What?

This PR aims to prevent the usage of the `order` CSS property.

## Why?

The `order` CSS property has a significant impact on accessibility and its usage must be prevented. In the very rare cases where its usage is appropriate, that should be a conscious choice of contributors and they should be required to add a disable comment with a disable reason.

## How?

- Adds the `order` CSS property to the `property-disallowed-list` rule.
- Adds an error message: `Avoid the order property. For accessibility reasons, visual, reading, and DOM order must match. Only use the order property when it does not affect reading order, meaning, and interaction`
- Makes a disable reason description required for all the `stylelint-disable*` comments by addint the config `"reportDescriptionlessDisables": true`.
- Adds a disable reason to all the existing disable comments.

## Testing Instructions




- Run `npm run lint:css` in your terminal, in the project root.
- Observe no errors are reported.
- Edit any .scss file in any package and add an order property e.g. `order: 1;`.
- Run `npm run lint:css` again.
- Observe stylelint reports an error.
- Check the disable reasons added to all `stylelint-disable` comments are meaningful and appropriate.

### Testing Instructions for Keyboard


## Screenshots or screencast
